.github/workflows/build.yml

```diff
@@ -143,10 +127,42 @@ jobs:
             - name: Run Psalm
               run: ./vendor/bin/psalm --output-format=github
 
+    build-phar:
+        needs: tests
+        runs-on: ubuntu-latest
+        name: Build PHAR file
+        steps:
+            - uses: actions/checkout@v3
+
+            - name: Setup PHP
+              uses: shivammathur/setup-php@v2
+              with:
+                  php-version: "8.3"
+                  ini-values: "phar.readonly=0,zend.exception_ignore_args=Off"
+                  coverage: none
+                  tools: box
+
+            - name: Build the PHAR
+              run: |
+                  composer config platform.php ^7.2.34 
+                  composer update --no-dev -o
+                  box build
+
+            - name: Test the PHAR
+              run: |
+                  behat.phar --version
```

Comment:
Two things:

* As it stands, this is still running in PHP 8.3 - should we run a second `setup-php` step to downgrade back to 7.2 to make sure the phar actually runs on the lowest supported version? 
* I *think* it would be straightforward here to actually run our behat suite (or perhaps tag a few features as `@phar-smoketest` or something) rather than just printing the version, which would give a lot more confidence the phar was working? It might be worth removing the `./vendor` directory first to be sure it's not taking anything from composer?

Reply:
Yes and Yes 😁